haystack/components/rankers/transformers_similarity.py

```diff
@@ -261,11 +267,28 @@ def run(
             text_to_embed = self.embedding_separator.join(meta_values_to_embed + [doc.content or ""])
             query_doc_pairs.append([self.query_prefix + query, self.document_prefix + text_to_embed])
 
-        features = self.tokenizer(query_doc_pairs, padding=True, truncation=True, return_tensors="pt").to(  # type: ignore
+        class _Dataset(Dataset):
+            def __init__(self, batch_encoding):
+                self.batch_encoding = batch_encoding
+
+            def __len__(self):
+                return len(self.batch_encoding["input_ids"])
+
+            def __getitem__(self, item):
+                return {key: self.batch_encoding.data[key][item] for key in self.batch_encoding.data.keys()}
+
+        batch_enc = self.tokenizer(query_doc_pairs, padding=True, truncation=True, return_tensors="pt").to(  # type: ignore
             self.device.first_device.to_torch()
         )
+        dataset = _Dataset(batch_enc)
+        inp_dataloader = DataLoader(dataset, batch_size=self.batch_size, shuffle=False)
+
+        similarity_scores = []
         with torch.inference_mode():
-            similarity_scores = self.model(**features).logits.squeeze(dim=1)  # type: ignore
+            for features in inp_dataloader:
+                model_preds = self.model(**features).logits.squeeze(dim=1)  # type: ignore
+                similarity_scores.extend(model_preds)
```

Comment:
Could this be wrapped in a try-except block for error handling?

Reply:
hmm how do you mean exactly. In case the model prediction fails? I don't think that's a design pattern we've used else where in our code base, but I'm open to the idea. We'd probably have to catch a generic Exception since I'm not sure what types of errors usually occur during model prediction. Also the error I'm trying to avoid here is Out of Memory errors which are not catchable in python.

Reply:
Oh interesting could you point me to the examples we have in other components?

Reply:
I don't know if its completely relevant but for instance [here in NEE component](https://github.com/deepset-ai/haystack/blob/675cf43be7f37ab83bf57b7cbe935756c1537735/haystack/components/extractors/named_entity_extractor.py#L157).

Reply:
ahh okay I gotcha, maybe something to consider for the future